File: 20-02-2024/aeroporto-lista-de-checagem/Aeroporto.java
import java.util.Scanner;

public class Aeroporto {
    
    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {
            System.out.println("Quantidade de passageiros:");

            final int amount = scanner.nextInt();
            scanner.nextLine();

            System.out.println("");

            for (int i = 0; i < amount; i++) {
                System.out.println("Passageiro (" + (i + 1) + ")");

                passenger(scanner);

                System.out.println("");
            }
        }
    }

    public static String get(Scanner scanner, String text) {
        System.out.println(text);

        return scanner.nextLine();
    }

    public static void passenger(Scanner scanner) {
        final String rg = get(scanner, "RG (ou 'Nao possui'):");

        if (rg.equalsIgnoreCase("Nao possui")) {
            System.out.println("A saída é nessa direção.");

            return;
        }

        final String birthdate = get(scanner, "Data de nascimento (RG):");
        final String ticket = get(scanner, "Passagem (ou 'Nao possui'):");

        if (ticket.equalsIgnoreCase("Nao possui")) {
            System.out.println("A recepção é nessa direção.");

            return;
        }

        final String ticketBirthdate = get(scanner, "Data de nascimento (Passagem):");

        if (!ticketBirthdate.equalsIgnoreCase(birthdate)) {
            System.out.println("190");

            return;
        }

        final String ticketSeat = get(scanner, "Assento (ex.: A12):");

        System.out.println("O seu assento é " + ticketSeat + ", tenha um ótimo dia.");
    }

}

File: 20-02-2024/aeroporto-lista-de-checagem/aeroporto.hs
import Control.Monad (when)
import Distribution.Compat.Prelude (exitSuccess)

getInput input = do
    putStrLn input
    getLine

pax n = do
    putStrLn ""
    putStrLn ("Passageiro (" ++ show n ++ ")")

    rg <- getInput "RG (ou 'Nao possui'):"

    when (rg == "Nao possui") $ do 
        putStrLn "A saída é nessa direção."
        exitSuccess

    birthdate <- getInput "Data de nascimento (RG):"

    ticket <- getInput "Passagem (ou 'Nao possui'):"

    when (ticket == "Nao possui") $ do
        putStrLn "A recepção é nessa direção."
        exitSuccess

    ticketBirthdate <- getInput "Data de nascimento (Passagem):"

    when (birthdate /= ticketBirthdate) $ do
        putStrLn "190"
        exitSuccess
    
    ticketSeat <- getInput "Assento (ex.: A12):"

    putStrLn ("O seu assento é " ++ ticketSeat ++ ", tenha um ótimo dia.")
    putStrLn ""

passengers current total
    | current == total = pure ()
    | otherwise = do
        pax (current + 1)
        passengers (current + 1) total

main :: IO () = do
    i <- getInput "Informe o número de passageiros: "

    let total = read i :: Int

    passengers 0 total